pkg/controller/controller.go

```diff
@@ -374,6 +376,66 @@ func (c *CSIControllerService) ListSnapshots(context.Context, *csi.ListSnapshots
 }
 
 // ControllerExpandVolume is not implemented yet
-func (c *CSIControllerService) ControllerExpandVolume(context.Context, *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented yet")
+func (c *CSIControllerService) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
+	ll := c.log.WithFields(logrus.Fields{
+		"method":   "ControllerExpandVolume",
+		"volumeID": req.GetVolumeId(),
+	})
+	ll.Infof("Processing request: %v", req)
+	var (
+		volID     = req.GetVolumeId()
+		lvg       = &lvgcrd.LVG{}
+		ctxWithID = context.WithValue(context.Background(), base.RequestUUID, volID)
+	)
+
+	if volID == "" {
+		return nil, status.Error(codes.InvalidArgument, "Volume name missing in request")
+	}
+	if req.GetVolumeCapability() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Volume capabilities missing in request")
+	}
+
+	volume := c.crHelper.GetVolumeByID(volID)
+
+	if volume == nil {
+		return nil, status.Error(codes.NotFound, "Volume doesn't exist")
+	}
+
+	if !util.IsStorageClassLVG(volume.Spec.StorageClass) {
+		return nil, status.Error(codes.FailedPrecondition, "Volume doesn't have LVG storage class")
+	}
+
+	if volume.Spec.Size > req.GetCapacityRange().GetRequiredBytes() {
```

Comment:
Should we use [`AlignSizeByPE`](https://github.com/dell/csi-baremetal/blob/06d06aebc80e7116d06497db2fad96bb75eac175/pkg/base/capacityplanner/common.go#L33 ) for `req.GetCapacityRange().GetRequiredBytes()`?